Doc/library/typing.rst

```diff
@@ -2815,6 +2815,56 @@ ABCs for working with IO
    represent the types of I/O streams such as returned by
    :func:`open`.
 
+The following protocols offer a simpler alternative for common use cases. They
+are especially useful for annotating function and method arguments and are
+decorated with :func:`@runtime_checkable <runtime_checkable>`.
+
+.. class:: Reader[T]
+
+   Protocol for reading from a file or other input stream.
+
+   .. method:: read(size=...)
```

Comment:
```suggestion
   .. method:: read(size=..., /)
```
(Same for other methods)

Reply:
I was unsure of how much of the signature to document. For example, would it also make sense to add argument and/or return types? I've added the slashes for now.